infra-as-code/bicep/acr.bicep

```diff
@@ -56,20 +82,41 @@ resource acrResource 'Microsoft.ContainerRegistry/registries@2023-07-01' = {
     zoneRedundancy: 'Enabled'
   }
 
-  // TODO (P2 - Jon): Add a build agent node connected to the virtual network, the image building subnet.
+  // TODO (P2 - Jen): Add a build agent node connected to the virtual network, the image building subnet.
   // Then pushes will happen from within the network when following the instructions in the README.
-  // TODO (P2 - Jon): I believe the user is going to need AcrPush in order to put the new image into ACR.
-
-  //resource x 'agentPools@2019-06-01-preview' = {
-  //  name: 'sdf'
-  //  properties: {
-  //    os: 'Linux'
-  //    count: 1
-  //    virtualNetworkSubnetResourceId: // TODO
-  //  }
-  //}
+  // TODO (P2 - Jen): I believe the user is going to need AcrPush in order to put the new image into ACR.
 }
 
+resource acrBuildAgentNode 'Microsoft.ContainerRegistry/registries/agentPools@2019-06-01-preview' = {
+  name: acrBuildAgentNodeName
+  location: location
+  parent: acrResource
+  properties: {
+    count: 1
+    os: 'Linux'
+    tier: linuxNodeAgentTier
+    virtualNetworkSubnetResourceId: acrSubnetID
+  }
+}
+
+resource acrPushRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(subscription().id, acrManagedIdentity.id, acrPushUserRole.id)
+  scope: acrResource
+  properties: {
+    roleDefinitionId: acrPushUserRole.id
+    principalId: acrManagedIdentity.properties.principalId
+  }
+}
+
+// ACR Pull Role Assignment
+resource acrPullRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(subscription().id, acrManagedIdentity.id, acrPullUserRole.id)
+  scope: acrResource
+  properties: {
+    roleDefinitionId: acrPullUserRole.id
+    principalId: acrManagedIdentity.properties.principalId
+  }
+}

```

Comment:
Can you let me know why it needs pull access in this scenario?

Reply:
I am not super familiar with the scenario - only wrapping up this one task. Ill remove it